src/Extensions.ts

```diff
@@ -55,5 +55,34 @@ export const extensions: Extension[] = [
     supportedMarkets: {
       '1_USDC_0xc3d688B66703497DAA19211EEdff47f25384cdc3': null
     }
+  },
+  {
+    id: 'collateral_swap',
+    name: 'Collateral Swap',
+    description: 'Enables swapping collateral on open loan positions without the need to repay the loan, in a single transaction',
+    developer: 'Wido Labs',
+    links: {
+      github: 'https://github.com/widolabs/compound-collateral-extension-ui',
+      website: 'https://www.joinwido.com/'
+    },
+    permissions: {
+      sign: '*',
+      trx: [
+        {
+          contract: '$operator',
+          abi: 'swapCollateral((address,uint256),(address,uint256),((uint8,bytes32,bytes32),(uint8,bytes32,bytes32)),(address,address,bytes),address)',
+          params: '*'
+        }
+      ]
```

Comment:
@kevincheng96 @hayesgm @mykelp  I have added the required permission for the app to execute the transaction. Could you verify that it looks fine? I am most interested in the `contract: '$operator'` line, since I'm not sure what the "$operator" placeholder is replaced with.

Reply:
the trx object does look ok to me. `$operator` is meant to signal your smart contract and you do look to have it specified correctly. What error are you currently getting?

Reply:
@torreyatcitty thanks, with this changes it should work fine then. 

Currently the issue is that the deployed version didn't include the `trx` permissions, so it's failing at the time of actually executing the transaction.

On the embedded sandbox it works fine, with or without the permissions, but once deployed it gets blocked without them.

Being this correct, I think next step would be to re-deploy so it takes the right permissions on the prod app?

Reply:
ah i see, yes the current version on prod does not have the `trx` permission. Will share an updated test build with the newer permission.